data/s3dis/indoor3d_util.py

```diff
@@ -0,0 +1,690 @@
+        data_batch[b, :, 1] -= (miny + block_size / 2)
+    new_data_batch[:, :, 0:6] = data_batch
+    return new_data_batch, label_batch
+
+
+def room2blocks_wrapper_normalized(data_label_filename,
+                                   num_point,
+                                   block_size=1.0,
+                                   stride=1.0,
+                                   random_sample=False,
+                                   sample_num=None,
+                                   sample_aug=1):
+    if data_label_filename[-3:] == 'txt':
+        data_label = np.loadtxt(data_label_filename)
+    elif data_label_filename[-3:] == 'npy':
+        data_label = np.load(data_label_filename)
+    else:
+        print('Unknown file type! exiting.')
+        exit()
+    return room2blocks_plus_normalized(data_label, num_point, block_size,
+                                       stride, random_sample, sample_num,
+                                       sample_aug)
+
+
+def room2samples(data, label, sample_num_point):
+    """Prepare whole room samples.
+
+    Args:
+        data: N x 6 numpy array, 012 are XYZ in meters, 345 are RGB in [0,1]
+            assumes the data is shifted (min point is origin) and
+            aligned (aligned with XYZ axis)
+        label: N size uint8 numpy array from 0-12
+        sample_num_point: int, how many points to sample in each sample
+    Returns:
+        sample_datas: K x sample_num_point x 9
+                     numpy array of XYZRGBX'Y'Z', RGB is in [0,1]
+        sample_labels: K x sample_num_point x 1 np array of uint8 labels
+    """
+    N = data.shape[0]
+    order = np.arange(N)
+    np.random.shuffle(order)
+    data = data[order, :]
+    label = label[order]
+
+    batch_num = int(np.ceil(N / float(sample_num_point)))
+    sample_datas = np.zeros((batch_num, sample_num_point, 6))
+    sample_labels = np.zeros((batch_num, sample_num_point, 1))
+
+    for i in range(batch_num):
+        beg_idx = i * sample_num_point
+        end_idx = min((i + 1) * sample_num_point, N)
+        num = end_idx - beg_idx
+        sample_datas[i, 0:num, :] = data[beg_idx:end_idx, :]
+        sample_labels[i, 0:num, 0] = label[beg_idx:end_idx]
+        if num < sample_num_point:
+            makeup_indices = np.random.choice(N, sample_num_point - num)
+            sample_datas[i, num:, :] = data[makeup_indices, :]
+            sample_labels[i, num:, 0] = label[makeup_indices]
+    return sample_datas, sample_labels
+
+
+def room2samples_plus_normalized(data_label, num_point):
+    """room2sample, with input filename and RGB preprocessing.
+
+    for each block centralize XYZ, add normalized XYZ as 678 channels
+    """
+    data = data_label[:, 0:6]
+    data[:, 3:6] /= 255.0
+    label = data_label[:, -1].astype(np.uint8)
+    max_room_x = max(data[:, 0])
+    max_room_y = max(data[:, 1])
+    max_room_z = max(data[:, 2])
+    # print(max_room_x, max_room_y, max_room_z)
```

Comment:
Remove unused comments (including others below).
Also make it more compact like comments above?